### PR TITLE
docs: Enable replica set with with docker-compose

### PR DESCRIPTION
**Pull request by rafaelsq**
## Description

I tried to run the project as described [here](https://github.com/appsmithorg/appsmith/blob/release/contributions/ServerSetup.md#steps-for-setup), however when trying to export and import a project I get a MongoDB error.

After asking for help on Discord [this PR](https://github.com/appsmithorg/appsmith/pull/20757) was merged.
It solves the problem, however it is a manual process.

So I create this PR to solve this problem.

## Type of change

- Bug fix (non-breaking change which fixes an issue)
- New feature (non-breaking change which adds functionality)
- Chore (housekeeping or task changes that don't impact user perception)

## How Has This Been Tested?

I deleted all project data and ran docker-compose.

### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

File: app/server/docker-compose.yml [hunk 1]
```diff
@@ -10,27 +10,44 @@ services:
...
+      redis:
+        condition: service_healthy
     networks:
       - appsmith
     volumes:
       - ./container-volumes/git-storage:/data/git-storage
 
   mongo:
-    image: mongo:4.4.6
+    image: mongo:6
```
Comment (AnaghHegde) [inline app/server/docker-compose.yml:23]: Can we make it v5 as the minimum version?

Comment (rafaelsq) [reply app/server/docker-compose.yml]: sure, done :heavy_check_mark:

**Comment by sharat87** [comment]
Hey @rafaelsq, sorry for the delay and not being able to merge this. We've now deprecated this `docker-compose.yml` file since it uses the `appsmith-server` image, which has been deprecated for a while now, and we've now also stopped building it.

So, I'll be closing this PR. Thank you for your time. 🙏

**Pull request closed without merging**